reference/Counter/CSharp-MVUX/Counter/MainModel.cs

```diff
@@ -1,11 +1,17 @@
 namespace Counter;
 
+internal partial record Countable(int Count, int Step)
+{
+    public Countable Increment() => this with
+    {
+        Count = Count + Step
+    };
+}
+
 internal partial record MainModel
 {
-    public IState<int> Count => State.Value(this, () => 0);
+    public IState<Countable> Countable => State.Value(this, () => new Countable(0, 1));
 
-    public IState<int> Step => State.Value(this, () => 1);
-    
-    public ValueTask IncrementCommand(int Step)
-            => Count.Update(c => c + Step, CancellationToken.None);
+    public ValueTask IncrementCommand()
```

Comment:
@dr1rrb Would be curious if there's anything you see here that we should do differently.